rclcpp_lifecycle/src/state.cpp

```diff
@@ -75,4 +102,48 @@ State::label() const
   return state_handle_->label;
 }
 
+void
+free_resources(State & instance)
+{
+  // nothing to free here
+  if (!instance.state_handle_) {
+    return;
+  }
+  if (!instance.owns_rcl_state_handle_) {
+    return;
+  }
+
+  if (instance.state_handle_) {
+    if (instance.state_handle_->label) {
+      instance.allocator_.deallocate(
+        const_cast<char *>(instance.state_handle_->label),
+        instance.allocator_.state);
+    }
+    instance.allocator_.deallocate(
+      const_cast<rcl_lifecycle_state_t *>(instance.state_handle_),
+      instance.allocator_.state);
+  }
+}
+
+void
+copy_from(State & lhs, const State & rhs)
+{
+  if (lhs.owns_rcl_state_handle_) {
+    free_resources(lhs);
+
+    lhs.allocator_ = rhs.allocator_;
+    lhs.owns_rcl_state_handle_ = rhs.owns_rcl_state_handle_;
+
+    auto state_handle = reinterpret_cast<rcl_lifecycle_state_t *>(
+      lhs.allocator_.allocate(sizeof(rcl_lifecycle_state_t), lhs.allocator_.state));
+    state_handle->id = rhs.state_handle_->id;
+    state_handle->label = rcutils_strndup(
+      rhs.state_handle_->label,
+      strlen(rhs.state_handle_->label),
+      lhs.allocator_);
+    lhs.state_handle_ = state_handle;
```

Comment:
Isn't `lhs` now owning the memory of `state_handle_`? But `lhs.owns_rcl_state_handle_` might be whatever `rhs.owns_rcl_state_handle_` was which could be `false.